backend/src/test/java/com/huellapositiva/integration/ManageNewsletterExcelActionShould.java

```diff
@@ -0,0 +1,72 @@
+package com.huellapositiva.integration;
+
+import com.huellapositiva.domain.actions.ManageNewsletterExcelAction;
+import com.huellapositiva.domain.model.valueobjects.EmailAddress;
+import com.huellapositiva.domain.service.EmailCommunicationService;
+import com.huellapositiva.domain.service.RemoteStorageService;
+import com.huellapositiva.infrastructure.orm.repository.JpaVolunteerRepository;
+import com.huellapositiva.util.TestData;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+
+import java.io.IOException;
+
+import static com.huellapositiva.util.TestData.DEFAULT_EMAIL;
+import static com.huellapositiva.util.TestData.DEFAULT_PASSWORD;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(MockitoExtension.class)
+@Import(TestData.class)
+class ManageNewsletterExcelActionShould {
+
+    @Autowired
+    private TestData testData;
+
+    @Autowired
+    private JpaVolunteerRepository jpaVolunteerRepository;
+
+    @MockBean
+    private EmailCommunicationService emailCommunicationService;
+
+    @MockBean
+    private RemoteStorageService remoteStorageService;
+
+    @BeforeEach
+    void beforeEach() {
+        testData.resetData();
+    }
+
+    @Test
+    void send_an_empty_newsletter_email_when_there_are_no_subscribed_volunteers() throws IOException {
+        // GIVEN
+        ManageNewsletterExcelAction manageNewsletterExcelAction = new ManageNewsletterExcelAction(
+                jpaVolunteerRepository, remoteStorageService, emailCommunicationService);
+
+        // WHEN
+        manageNewsletterExcelAction.execute(DEFAULT_EMAIL);
+
+        // THEN
+        verify(emailCommunicationService).sendEmptyNewsletter(EmailAddress.from(DEFAULT_EMAIL));
+    }
+
+    @Test
+    void send_a_newsletter_email_when_there_are_subscribed_volunteers() throws IOException {
+        testData.createSubscribedVolunteer(DEFAULT_EMAIL, DEFAULT_PASSWORD);
+        // GIVEN
+        ManageNewsletterExcelAction manageNewsletterExcelAction = new ManageNewsletterExcelAction(
+                jpaVolunteerRepository, remoteStorageService, emailCommunicationService);
+
+        // WHEN
+        manageNewsletterExcelAction.execute(DEFAULT_EMAIL);
+
+        // THEN
+        verify(emailCommunicationService).sendNewsletter(EmailAddress.from(DEFAULT_EMAIL),null);
+    }
+}
```

Comment:
Unit test or integration test?

Reply:
Ok I changed it from integration to unit